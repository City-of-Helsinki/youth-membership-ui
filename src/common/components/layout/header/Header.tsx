import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import styles from './Header.module.css';
import LanguageSwitcher from '../../../../i18n/languageSwitcher/LanguageSwitcher';
import FullscreenNavigation from '../../fullscreenNavigation/FullscreenNavigation';
import HelsinkiLogo from '../../helsinkiLogo/HelsinkiLogo';
import UserDropdown from './userDropdown/UserDropdown';

// Approver variant is shown for approver view. The approver should not
// be shown the user menu, nor should the header contain links for
// navigating into te front page.
export type HeaderVariant = 'default' | 'approver';

type Props = {
  variant?: HeaderVariant;
};

function Header({ variant = 'default' }: Props) {
  const { t } = useTranslation();
  const isDefaultVariant = variant === 'default';
  const isApproverVariant = variant === 'approver';

  return (
    <header className={styles.header}>
      <div className={styles.content}>
        <HelsinkiLogo
          className={styles.logo}
          isLinkToFrontPage={isDefaultVariant}
        />
        {isDefaultVariant && (
          <Link to="/" className={styles.appName}>
            {t('appName')}
          </Link>
        )}
        {isApproverVariant && (
          <span className={styles.appName}>{t('appName')}</span>
        )}
        <section className={styles.end}>
          <FullscreenNavigation className={styles.mobileNav} />
          <div className={styles.desktopNav}>
            <LanguageSwitcher />
            {isDefaultVariant && <UserDropdown />}
          </div>
        </section>
      </div>
    </header>
  );
}

export default Header;
