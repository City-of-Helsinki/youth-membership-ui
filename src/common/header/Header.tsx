import React from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import styles from './Header.module.css';
import LanguageSwitcher from '../../i18n/languageSwitcher/LanguageSwitcher';
import FullscreenNavigation from '../fullscreenNavigation/FullscreenNavigation';
import UserDropdown from './userDropdown/UserDropdown';
import HelsinkiLogo from '../helsinkiLogo/HelsinkiLogo';

type Props = {};

function Header(props: Props) {
  const { t } = useTranslation();
  return (
    <header className={styles.header}>
      <div className={styles.centeredContainer}>
        <HelsinkiLogo className={styles.logo} isLinkToFrontPage />
        <Link to="/" className={styles.appName}>
          {t('appName')}
        </Link>
        <section className={styles.end}>
          <FullscreenNavigation className={styles.mobileNav} />
          <div className={styles.desktopNav}>
            <LanguageSwitcher />
            <UserDropdown />
          </div>
        </section>
      </div>
    </header>
  );
}

export default Header;
