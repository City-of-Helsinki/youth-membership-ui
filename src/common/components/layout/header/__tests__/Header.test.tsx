import React from 'react';
import toJson from 'enzyme-to-json';

import { mountWithProvider } from '../../../../test/testUtils';
import Header from '../Header';

test.skip('matches snapshot', () => {
  const wrapper = mountWithProvider(<Header />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
