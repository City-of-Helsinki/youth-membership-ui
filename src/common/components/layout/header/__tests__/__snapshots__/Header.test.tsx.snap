// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
>
  <div
    className="content"
  >
    <HelsinkiLogo
      className="logo"
      isLinkToFrontPage={true}
    />
    <Link
      className="appName"
      to="/"
    >
      Jässäri
    </Link>
    <section
      className="end"
    >
      <FullscreenNavigation
        className="mobileNav"
      />
      <div
        className="desktopNav"
      >
        <LanguageSwitcher />
        <UserDropdown />
      </div>
    </section>
  </div>
</header>
`;
